drivers/include/dose.h

```diff
@@ -137,13 +155,23 @@ typedef enum {
  *  Fallback to idle if the remote side died within a transaction.
  */
 #ifndef CONFIG_DOSE_TIMEOUT_BYTES
-#define CONFIG_DOSE_TIMEOUT_BYTES       (50)
+#define CONFIG_DOSE_TIMEOUT_BYTES   (50)
 #endif
 /** @} */
 
 #define DOSE_FRAME_CRC_LEN          (2)     /**< CRC16 is used */
 #define DOSE_FRAME_LEN (ETHERNET_FRAME_LEN + DOSE_FRAME_CRC_LEN) /**< dose frame length */
 
+/**
+ * @brief   Hardware timer to use with the `dose_watchdog` module.
+ *
+ *          This will be used to detect RX timeout instead of xtimer to speed up
+ *          the RX path when high data rates / less CPU overhead is required.
+ */
+#if DOXYGEN && !defined(DOSE_TIMER_DEV)
+#define DOSE_TIMER_DEV              TIMER_DEV(1)
```

Comment:
I don't think that drivers should place defaults; I'm already unhappy with sys doing defaults (which, for timers, happens for xtimer only), but a driver picking a default IMO runs too high a risk of the board or application developer overlooking it and using the peripheral twice.

```suggestion
#if DOXYGEN
#define DOSE_TIMER_DEV              TIMER_DEV(...)
#elif !defined(DOSE_TIMER_DEV)
#error "DOSE_TIMER_DEV needs to be set by the board (if preconfigured) or application, eg. to TIMER_DEV(1)"
```

(Not putting a hard block here if it's something we generally do).